### DISCUSSION TITLE
Best practice for navigating after action

### DISCUSSION THREAD

**Discussion by stevepryde** [leptos-rs/leptos#2930] [General]
If I call `navigate()` from within `create_action()` I get a warning in the browser console that looks like this:

```
Attempted to update a signal after it was disposed.
signal created here: /Users/steve/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_server-0.6.14/src/action.rs:153:23
warning happened here: /Users/steve/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_server-0.6.14/src/action.rs:401:29
```

And also this:
```
[Signal::update] At Location { file: "/Users/steve/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_server-0.6.14/src/action.rs", line: 401, col: 29 }, you’re trying to update a Signal<usize> (defined at /Users/steve/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_server-0.6.14/src/action.rs:153:23) that has already been disposed of. This is probably a logic error in a component that creates and disposes of scopes. If it does not cause any issues, it is safe to ignore this warning, which occurs only in debug mode.
```

The same thing happens if I set the value of a signal which then causes the component containing the `create_action()` to be disposed, so it's not strictly related to `navigate()`.

It seems to only happen within `create_action()`. If I instead use `spawn_local()` I don't get that warning.

Is this a bogus warning in this instance? Or is there something I should be doing instead to avoid the warning?

**Comment by gbj** [comment]
If I'm understanding correctly it's just a warning and not an error, and everything is working fine? In your case it is basically a false positive: you *know* that you are setting a signal that has already been disposed and that is fine. (It is because navigating away from the page disposes of the action while the action is still in flight, and then when the action resolves it tries to update its `value` signal but it's no longer there.) 

EDIT: It does raise some reasonable questions — For example, does it matter if your action returns successfully or is an error? You have no way of knowing which it is. So I don't think it's a totally useless warning/false positive in that sense.

**Reply by stevepryde** [reply]
Well I have it set up so that it only navigates on success, so I think it is a false positive here. That's what I thought but I wanted to check that there wasn't something special about the inner workings of actions that makes this a bad thing to do. Thanks for the clarification. That makes sense regarding the update of its `value` signal.

Is there any way to silence this warning for this use case?

**Comment by benwis** [comment]
If you call get_untracked() instead of get() it should go away
